src/DynExpManager/Instruments/NP_Conex_CC.h

```diff
@@ -0,0 +1,344 @@
+		virtual bool HasFailedChild() const noexcept override;
+
+		Conex_CCStatusType Conex_CCStatus;
+		ErrorCodeType ErrorCode = NoError;
+		size_t NumFailedStatusUpdateAttempts = 0;
+	};
+
+	class NP_Conex_CC_Params : public PositionerStageParams
+	{
+	public:
+		NP_Conex_CC_Params(DynExp::ItemIDType ID, const DynExp::DynExpCore& Core) : PositionerStageParams(ID, Core) {}
+		virtual ~NP_Conex_CC_Params() = default;
+
+		virtual const char* GetParamClassTag() const noexcept override { return "NP_Conex_CC_Params"; }
+
+		Param<DynExp::ObjectLink<DynExp::SerialCommunicationHardwareAdapter>> HardwareAdapter = { *this, GetCore().GetHardwareAdapterManager(),
+			"HardwareAdapter", "Serial port", "Underlying hardware adapter of this instrument", DynExpUI::Icons::HardwareAdapter };
+		Param<ParamsConfigDialog::NumberType> ConexAddress = { *this, "ConexAddress", "Conex address",
+			"Address (1-31) of the Conex controller to be used", true, 1, 31 };
+
+	private:
+		void ConfigureParamsImpl(dispatch_tag<PositionerStageParams>) override final { ConfigureParamsImpl(dispatch_tag<NP_Conex_CC_Params>()); }
+		virtual void ConfigureParamsImpl(dispatch_tag<NP_Conex_CC_Params>) {}
+	};
+
+	class NP_Conex_CC_Configurator : public PositionerStageConfigurator
+	{
+	public:
+		using ObjectType = NP_Conex_CC;
+		using ParamsType = NP_Conex_CC_Params;
+
+		NP_Conex_CC_Configurator() = default;
+		virtual ~NP_Conex_CC_Configurator() = default;
+
+	private:
+		virtual DynExp::ParamsBasePtrType MakeParams(DynExp::ItemIDType ID, const DynExp::DynExpCore& Core) const override { return DynExp::MakeParams<NP_Conex_CC_Configurator>(ID, Core); }
+	};
+
+	class NP_Conex_CC : public PositionerStage
+	{
+	public:
+		using ParamsType = NP_Conex_CC_Params;
+		using ConfigType = NP_Conex_CC_Configurator;
+		using InstrumentDataType = NP_Conex_CCStageData;
+
+		static std::string AnswerToNumberString(std::string&& Answer, const char* StartCode);
+
+		constexpr static auto Name() noexcept { return "NP Conex-CC"; }
+
+		NP_Conex_CC(const std::thread::id OwnerThreadID, DynExp::ParamsBasePtrType&& Params);
+		virtual ~NP_Conex_CC() {}
+
+		virtual std::string GetName() const override { return Name(); }
+
+		virtual PositionerStageData::PositionType GetMinPosition() const noexcept override { return -180e6; }
+		virtual PositionerStageData::PositionType GetMaxPosition() const noexcept override { return 180e6; }
+		virtual PositionerStageData::PositionType GetResolution() const noexcept override { return 1; }
+		virtual PositionerStageData::PositionType GetMinVelocity() const noexcept override { return 0; }
+		virtual PositionerStageData::PositionType GetMaxVelocity() const noexcept override { return 1e17; }
+		virtual PositionerStageData::PositionType GetDefaultVelocity() const noexcept override { return 10e6; }
+		virtual double GetFloatToIntConversion() const noexcept { return 1e6; } // the controller expects a float as position with 6 digits of precision
```

Comment:
Why is this function public? Why is it virtual? Please chose a descriptive name.